lib/ZwaveClient.ts

```diff
@@ -0,0 +1,3276 @@
+	}
+
+	async abortFirmwareUpdate(nodeId: number) {
+		if (this.driver && !this.closed) {
+			const zwaveNode = this.getNode(nodeId)
+
+			if (!zwaveNode) {
+				throw Error(`Node ${nodeId} not found`)
+			}
+
+			return zwaveNode.abortFirmwareUpdate()
+		}
+
+		throw Error('Driver is closed')
+	}
+
+	beginHealingNetwork(): boolean {
+		if (this.driver && !this.closed) {
+			return this.driver.controller.beginHealingNetwork()
+		}
+
+		throw Error('Driver is closed')
+	}
+
+	stopHealingNetwork(): boolean {
+		if (this.driver && !this.closed) {
+			return this.driver.controller.stopHealingNetwork()
+		}
+
+		throw Error('Driver is closed')
+	}
+
+	async hardReset() {
+		if (this.driver && !this.closed) {
+			return this.driver.hardReset()
+		}
+
+		throw Error('Driver is closed')
+	}
+
+	/**
+	 * Send a command
+	 */
+	async sendCommand(
+		ctx: {
+			nodeId: number
+			endpoint: number
+			commandClass: CommandClasses | keyof typeof CommandClasses
+		},
+		command: string,
+		args: any[]
+	): Promise<any> {
+		if (this.driver && !this.closed) {
+			if (typeof ctx.nodeId !== 'number') {
+				throw Error('nodeId must be a number')
+			}
+
+			if (args !== undefined && !Array.isArray(args)) {
+				throw Error('if args is given, it must be an array')
+			}
+
+			// get node instance
+			const node = this.getNode(ctx.nodeId)
+			if (!node) {
+				throw Error(`Node ${ctx.nodeId} was not found!`)
+			}
+
+			// get the endpoint instance
+			const endpoint = node.getEndpoint(ctx.endpoint || 0)
+			if (!endpoint) {
+				throw Error(
+					`Endpoint ${ctx.endpoint} does not exist on Node ${ctx.nodeId}!`
+				)
+			}
+
+			const commandClass =
+				typeof ctx.commandClass === 'number'
+					? ctx.commandClass
+					: CommandClasses[ctx.commandClass]
+
+			// get the command class instance to send the command
+			const api = endpoint.commandClasses[commandClass]
+			if (!api || !api.isSupported()) {
+				throw Error(
+					`Node ${ctx.nodeId} (Endpoint ${ctx.endpoint}) does not support CC ${ctx.commandClass} or it has not been implemented yet`
+				)
+			} else if (!(command in api)) {
+				throw Error(
+					`The command ${command} does not exist for CC ${ctx.commandClass}`
+				)
+			}
+
+			// send the command with args
+			const method = api[command].bind(api)
+			const result = args ? await method(...args) : await method()
+			return result
+		}
+
+		throw Error('Driver is closed')
+	}
+
+	/**
+	 * Calls a specific `client` or `ZwaveClient` method based on `apiName`
+	 * ZwaveClients methods used are the ones that overrides default Zwave methods
+	 * like nodes name and location and scenes management.
+	 *
+	 * @param {string} apiName The api name
+	 * @param {any[]} args Array of arguments to use for the api call
+	 * @returns An object `{success: <success>, message: <message>, args: <args>, result: <the response>}`,  if success is false the message contains the error
+	 */
+	async callApi(apiName: string, ...args: any[]) {
```

Comment:
Clearly advanced territory, but there is a way to define this in a typesafe way:
https://www.typescriptlang.org/play?#code/C4TwDgpgBAwghgGwQQQAoEkBKEDOBXBYAHgBUA+KAXigG8AoKKfAY2dxwC4pgAnPCADQMoAW3ZwA5hC45eASwB2EoYx64CwLiToBfKAB9awlmxycoAM0Q5BwsWcnSm8pbrp1mCOGagAxAPb+RowWgQAUcDwSAIwyLhIAlFwKeCIARhA8wYxQasB4PApQ0QDcwjrujGmREVEATMmpGTxJUGmBCBBwRfQ5uRD5hdx8EGWMFcLMiChgcqRQEAAewBAKACY4UACii554axBEANYQIP4WfoECUABEU0jIszdkZGEKcGJa1wB0v5ES5lQkQ+A0yOCIAX8AG0SABdMiteAPDDYfCEIjYQYKEjgQ6QmHwii9HJ5Ao9YR9EzsLi8fgqPqicRSLg3fxHG70vpqNGaKBhYAACzkm28UG6IASUPeYlhYV+33+OASFKgFXGbg8-gUsksgSoUAUEAA7pd-GEEmVmFqdXl9aF-N97jM5GEbvaObcbhbNdr-J1vgh-BIwnkLUA

Reply:
Nopt working, see slack